tests/helpers.py

```diff
@@ -0,0 +1,147 @@
+
+    if name is None:
+        name = "table-%s" % uuid.uuid4()
+
+    table = SqlaTable(
+        table_name=name,
+        #
+        schema=schema,
+        params=json.dumps(dict(
+            tags=tags,
+            database_name=database_name,
+        )),
+    )
+
+    # Return imported obj
+    return superset.db.session.query(SqlaTable).filter_by(
+        id=SqlaTable.import_obj(table)).first()
+
+
+def new_slice(name=None, table=None, tags=['test']):
+
+    """Create a new test slice (and test table if none specified)"""
+
+    if name is None:
+        name = "slice-%s" % uuid.uuid4()
+
+    if table is None:
+        table = create_table(tags=tags)
+
+    slyce = Slice(
+        slice_name=name,
+        datasource_type='table',
+        datasource_name=table.datasource_name,
+        viz_type='bubble',
+        params=json.dumps(dict(
+            tags=tags,
+            database_name=table.database_name,
+            datasource_name=table.datasource_name,
+            schema=table.schema,
+            metrics=[],
+        )),
+    )
+
+    # NOTE that we don't actually import the slice here - it needs to
+    # be attached to a dashboard for that to make sense
+    return slyce
+
+
+def create_dashboard(title=None, slices=None, tags=['test']):
+
+    """Create a new test dashboard (and slice and table if needed)"""
+
+    if title is None:
+        title = "dashboard-%s" % uuid.uuid4()
+
+    if slices is None:
+        slices = [new_slice(tags=tags)]
+
+    dashboard = Dashboard(
+        dashboard_title=title,
+        slug=title.lower(),
+        slices=slices,
+        position_json=json.dumps(dict(size_y=2, size_x=2)),
+        json_metadata=json.dumps(dict(tags=tags)),
+    )
+
+    # Return imported obj
+    return superset.db.session.query(Dashboard).filter_by(
+        id=Dashboard.import_obj(dashboard)).first()
+
+
+def cleanup_data(tags=['test']):
+    cleanup_objs(Dashboard, tags)
+    cleanup_objs(Slice, tags)
+    cleanup_objs(SqlaTable, tags)
+
+
+def cleanup_objs(model, tags=['test']):
+
+    """
+    Delete datas that are tagged with certain things
+    TODO: Think about tracking with a fixture obj instead, cleaner?
+    """
+
+    import celery
+    logger = celery.utils.log.get_task_logger(__name__)
+
+    tags = set(tags)
+    for obj in superset.db.session.query(model):
+        obj_tags = getattr(obj, 'params_dict', {}).get('tags', [])
+        if tags.isdisjoint(obj_tags):
+            continue
+        superset.db.session.delete(obj)
+    superset.db.session.commit()
+
+def grant_db_access_to_role(role, db):
+    return grant_obj_permission_to_role(role, db, 'database_access')
+
+
+def grant_slice_access_to_role(role, slyce):
+    return grant_obj_permission_to_role(role, slyce, 'datasource_access')
+
+
+def grant_obj_permission_to_role(role, obj, permission):
+
+    """
+    Does the extremely confusing steps required to add permission for a role
```

Comment:
For real, this was no fun to look at.  Worth a second set of eyes because naming is pretty confusing - basically all permissions in superset are tied to "view menus" or "edit menus" and such, which is weird when we're making an API that doesn't use those menus.